pkg/rtc/participant.go

```diff
@@ -459,6 +460,44 @@ func (p *ParticipantImpl) SetMetadata(metadata string) {
 	}
 }
 
+func (p *ParticipantImpl) SetAttributes(attrs map[string]string) error {
+	p.lock.Lock()
+	grants := p.grants.Load().Clone()
+	if grants.Attributes == nil {
+		grants.Attributes = make(map[string]string)
+	}
+	for k, v := range attrs {
+		grants.Attributes[k] = v
+	}
```

Comment:
did we want to treat set with a null string as delete?

Reply:
sure, I can add that